### PR TITLE
Create Frogbot config schema

### PR DESCRIPTION
**Pull request by yahavi**
- [x] All [tests](https://github.com/jfrog/frogbot#tests) passed. If this feature is not already covered by the tests, I added new tests.
- [x] This pull request is on the dev branch.
- [x] I used gofmt for formatting the code before submitting the pull request.
---

Create the schema of the `frogbot-config.yml` file.
This schema will be loaded automatically on every modern IDE, after adding it to the SchemaStore later on.

**Examples:**

* Create Frogbot configuration from scratch:
image

* Provide explanation on hovering:
image

* Config validation:
image

### CODE REVIEW

File: schema/frogbotschema_test.go [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+		// Read frogbot config
+		frogbotConfigFile, err := os.ReadFile(frogbotConfigFilePath)
+		assert.NoError(t, err)
+
+		// Unmarshal frogbot config
+		var frogbotConfigYaml interface{}
+		err = yaml.Unmarshal(frogbotConfigFile, &frogbotConfigYaml)
+		assert.NoError(t, err)
+
+		// Convert the Yaml schema to JSON schema to help the json parser to validate it
```
Comment (eyalbe4) [inline schema/frogbotschema_test.go:67]: ...json parser to validate it --> ...json parser validate it.

File: schema/frogbotschema_test.go [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+		{"no-repo", "repoName is required"},
+		{"empty-repo", "Expected: string, given: null"},
+	}
+	for _, testCase := range testCases {
+		validateSchema(t, schemaLoader, filepath.Join("testdata", testCase.testName+".yml"), testCase.errorString)
+	}
+}
+
+// Validate frogbot config against the frogbot schema
+func validateSchema(t *testing.T, schemaLoader gojsonschema.JSONLoader, frogbotConfigFilePath, expectError string) {
```
Comment (eyalbe4) [inline schema/frogbotschema_test.go:56]: Let's describe the arguments this function receives, and specifically the `expectError` argument.

File: schema/frogbotschema_test.go [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+			assert.False(t, result.Valid())
+			assert.Contains(t, result.Errors()[0].String(), expectError)
+		} else {
+			assert.True(t, result.Valid(), result.Errors())
+		}
+	})
+}
+
+// Recursively convert yaml interface to JSON interface
+func convertYamlToJson(yamlValue interface{}) interface{} {
```
Comment (eyalbe4) [inline schema/frogbotschema_test.go:84]: Wouldn't be easier to unmarshal the YAML into a string, and then marshal the string into a JSON?

**Review by eyalbe4 — CHANGES_REQUESTED** [review]

File: schema/frogbotschema_test.go (hunk 3 above)
Comment (yahavi) [reply schema/frogbotschema_test.go:84]: I'm not sure I understand your suggestion but here is the current process:
`YAML bytes` -> `YAML object` -> `JSON object` -> `JSON validation`

The only way I can think we can convert YAML to JSON without `convertYamlToJson` is to convert the YAML bytes into [FrogbotRepoConfig](https://github.com/jfrog/frogbot/blob/v2.5.5/commands/utils/params.go#L30) struct. However, the config contains many fields without `omitempty`. Fixing that requires a big config refactor which is out of the scope of this PR.

**Pull request merged**